Index: django/db/models/base.py
===================================================================
--- django/db/models/base.py	(revision 9108)
+++ django/db/models/base.py	(working copy)
@@ -48,7 +48,8 @@
             # Figure out the app_label by looking one level up.
             # For 'django.contrib.sites.models', this would be 'sites'.
             model_module = sys.modules[new_class.__module__]
-            kwargs = {"app_label": model_module.__name__.split('.')[-2]}
+            module_names = model_module.__name__.split('.')
+            kwargs = {"app_label": module_names[module_names.index('models')-1]}
         else:
             kwargs = {}
 
